File: frontend/src/api/image.ts
import axios, { AxiosError } from 'axios'
import api from './api'

export interface ImageGenerationResponse {
  image_url: string
}

interface ErrorResponse {
  detail?: string
}

function messageFor(err: AxiosError<ErrorResponse>): string {
  const status = err.response?.status
  if (status === 503) return '图像生成服务暂时不可用，已自动重试仍失败，请稍后再试'
  if (status === 422) return '描述内容不符合要求：不能为空，且不超过 1000 字'
  if (status != null) return '生成失败，请重试'
  return '无法连接服务器，请确认后端服务已启动'
}

export async function generateImage(prompt: string): Promise<string> {
  try {
    const response = await api.post<ImageGenerationResponse>('/image/generate', { prompt })
    return response.data.image_url
  } catch (err) {
    if (axios.isAxiosError<ErrorResponse>(err)) {
      throw new Error(messageFor(err))
    }
    throw err
  }
}

File: backend/app/routers/image.py
"""Image generation endpoint."""

import logging

from fastapi import APIRouter, Depends, HTTPException

from app.config import get_settings
from app.dependencies import get_current_user
from app.models import User
from app.schemas import ImageGenerationRequest, ImageGenerationResponse
from app.services.image_service import ImageGenerationService

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/image", tags=["image"])


def get_image_service() -> ImageGenerationService:
    """Dependency: image generation service."""
    settings = get_settings()
    if not settings.image_api_key:
        raise HTTPException(status_code=503, detail="Image generation not configured")
    return ImageGenerationService(settings)


@router.post("/generate", response_model=ImageGenerationResponse)
async def generate_image(
    request: ImageGenerationRequest,
    user: User = Depends(get_current_user),
    service: ImageGenerationService = Depends(get_image_service),
) -> ImageGenerationResponse:
    """Generate an image from a text prompt for an authenticated user."""
    try:
        image_url = await service.generate_image(request.prompt)
    except Exception as exc:
        logger.exception("Image generation failed")
        raise HTTPException(status_code=502, detail="Image generation failed") from exc
    return ImageGenerationResponse(image_url=image_url)
